firmware/hw_layer/hardware.cpp

```diff
@@ -264,9 +264,20 @@ void stopSpi(spi_device_e device) {
  * this method is NOT currently invoked on ECU start
  * todo: maybe start invoking this method on ECU start so that peripheral start-up initialization and restart are unified?
  */
+
 void applyNewHardwareSettings(void) {
     // all 'stop' methods need to go before we begin starting pins
 
+	PointerListNode *listItem = &buttonDebounceListHead;
```

Comment:
an argument could be made that we already use #include "utlist.h" which gives is list utilities for example LL_FOREACH2 see event_queue.cpp

I do not have an opinion on if this should be used here or not. Probably not since we have such a trivial use-case? @mck1117 ?

Reply:
I don't think we need it for this use case either.

What we should do is move this code (lines 271-279) inside a static function on `ButtonDebounce`:

```
class ButtonDebounce {
    static void updateConfigurationList();

private:
    static ButtonDebounce* s_firstDebounce = nullptr;
};
```

Reply:
Are you saying s_firstDebounce should replace buttonDebounceListHead? If so, why should it be a pointer?

Reply:
Oh wait no it's ButtonDebounce type. I was thinking PointerListNode for some reason. never mind.

Reply:
I could get rid of PointerListNode entirely and just store the 'next' pointer in ButtonDebounce. Is that what you're thinking?